Fatras/include/ActsFatras/Kernel/Simulation.hpp

```diff
@@ -284,12 +291,13 @@ struct Simulation {
     // initial particle state was already pushed to the container before
     // store final particle state at the end of the simulation
     particlesFinal.push_back(result.particle);
+    std::copy(result.hits.begin(), result.hits.end(), std::back_inserter(hits));
```

Comment:
This is just to order the code more logically and does not change functionality, right?

Reply:
Yes - my tough was to group the copying to output stuff together and later handle the produced secondaries which get pushed into the initial vector